snakemake/cli.py

```diff
@@ -1849,41 +1844,16 @@ def parse_quietness(quietness) -> Set[Quietness]:
 
 
 def setup_log_handlers(args, parser):
-    log_handler = []
-    if args.log_handler_script is not None:
-        if not os.path.exists(args.log_handler_script):
-            print(
-                "Error: no log handler script found, {}.".format(
-                    args.log_handler_script
-                ),
-                file=sys.stderr,
-            )
-            sys.exit(1)
-        log_script = SourceFileLoader("log", args.log_handler_script).load_module()
-        try:
-            log_handler.append(log_script.log_handler)
-        except:
-            print(
-                'Error: Invalid log handler script, {}. Expect python function "log_handler(msg)".'.format(
-                    args.log_handler_script
-                ),
-                file=sys.stderr,
-            )
-            sys.exit(1)
+    log_handlers = []
 
-    if args.log_service == "slack":
-        slack_logger = logging.SlackLogger()
-        log_handler.append(slack_logger.log_handler)
-
-    elif args.wms_monitor or args.log_service == "wms":
-        # Generate additional metadata for server
-        metadata = generate_parser_metadata(parser, args)
-        wms_logger = logging.WMSLogger(
-            args.wms_monitor, args.wms_monitor_arg, metadata=metadata
-        )
-        log_handler.append(wms_logger.log_handler)
+    for logger_arg in args.logger:
+        plugin = LoggerPluginRegistry().get_plugin(plugin_name=logger_arg)
+        plugin_settings = plugin.get_settings(args)
+        # logger plugin here is the subclass of LoggerPluginBase, we get handler from it in configure_logger.
+        logger_plugin = plugin.logger_plugin(plugin_settings)
```

Comment:
It seems confusing that the logger plugin returns a LoggerPluginBase subclass. Should that be called LogHandlerBase, and the method `plugin.log_handler(plugin_settings)`?

Reply:
I was confused looking at this again, so agreed. Just so we're on the same page you're suggesting changing this in: `src/snakemake_interface_logger_plugins/registry/plugin.py`


```diff
 @dataclass
 class Plugin(PluginBase):
-    logger_plugin: object
+    log_handler: object
     _logger_settings_cls: Optional[Type[LoggerPluginSettingsBase]]
     _name: str
```

And then in `src/snakemake_interface_logger_plugins/base.py`

```diff
-class LoggerPluginBase(ABC):
+class LogHandlerBase(ABC):
     def __init__(
         self,
         settings: Optional[LoggerPluginSettingsBase],
     ) -> None:
         self.settings = settings
         self.__post__init()
```

Reply:
Yes. In addition, I would use `log_handler: LogHandlerBase` as type annotation in `src/snakemake_interface_logger_plugins/registry/plugin.py`.

Reply:
Resolved in https://github.com/cademirch/snakemake-interface-logger-plugins/commit/2b23736a53a569d2cf9fc702d4eed39d283c39bd